File: HoneyStoreUI/src/app/models/cart-item.ts
import { Product } from './product';

export class CartItem {
    id: number;
    productId: number;
    product: Product;
    userId: number | undefined;
    quantity: number;
    createdOn: Date | undefined;    
    orderId: number | null;

    constructor(productId: number = 1, userId: number = 1, quantity: number = 1) {
        this.id = 0;
        this.productId = productId;
        this.product = new Product();
        this.userId = userId;   
        this.quantity = quantity;
        this.createdOn = new Date();               
        this.orderId = null;
    }
}

File: HoneyStore.DataAccess/Entities/CartItem.cs
using HoneyStore.DataAccess.Interfaces;

namespace HoneyStore.DataAccess.Entities
{
    public class CartItem: IIdentifier
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public int Quantity { get; set; }
        
        public DateTime CreatedOn { get; set; }

        public int ProductId { get; set; }

        public Product Product { get; set; }

        public int? OrderId { get; set; }
    }
}